src/roswire/ros2/source.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+__all__ = ("ROS2PackageSourceExtractor",)
+
+import os.path
+import typing as t
+
+import attr
+import dockerblade
+from loguru import logger
+
+from .. import AppInstance
+from ..common.source import (
+    extract_sources_from_cmake,
+    NodeSourceInfo,
+    PackageSourceExtractor
+)
+
+
+@attr.s(auto_attribs=True)
+class ROS2PackageSourceExtractor(PackageSourceExtractor):
+    _files: dockerblade.FileSystem
+
+    @classmethod
+    def for_app_instance(
+        cls,
+        app_instance: AppInstance
+    ) -> "ROS2PackageSourceExtractor":
+        return ROS2PackageSourceExtractor(files=app_instance.files)
+
+    def extract_source_for_package(
+        self,
+        path_to_package: str
+    ) -> t.Mapping[str, NodeSourceInfo]:
+        cmakelists_path = os.path.join(path_to_package, "CMakeLists.txt")
+
+        if self._files.isfile(cmakelists_path):
+            contents = self._files.read(cmakelists_path)
+            source_infos = extract_sources_from_cmake(contents)
```

Comment:
e.g., you would call `self._extract_sources_from_cmake` here. `_extract_sources_from_cmake` can be defined as a static or class method in `PackageSourceExtractor`.